zerver/views/registration.py

```diff
@@ -532,3 +533,33 @@ def realm_redirect(request: HttpRequest) -> HttpResponse:
         form = RealmRedirectForm()
 
     return render(request, 'zerver/realm_redirect.html', context={'form': form})
+
+@csrf_exempt
+def register_development_user(request: HttpRequest) -> HttpResponseRedirect:
+    realm = get_realm("zulip")
```

Comment:
I think that if you're on the page for a given realm, we should use that one.

I think using `get_realm_from_request` (and mapping `None` => `get_realm("zulip")) for this is the right approach